### PR TITLE
Documentation for SpaceGame class

### PR DESCRIPTION
**Pull request by Grace0Hud**
I commented SpaceGame.java from what I understood of the code.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <component name="ProjectKey">
     <option name="state" value="project://e79810c8-c5c8-43b1-b19c-90c1f4095425" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_15_PREVIEW" default="false" project-jdk-name="15" project-jdk-type="JavaSDK">
+   <component name="ProjectRootManager" version="2" languageLevel="JDK_15_PREVIEW" default="false" project-jdk-name="15" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/out" />
```
Comment (hellminister) [inline .idea/misc.xml]: please undo this change, you can set it so the modifications of this file doesnt get commited

File: src/spacegame3/SpaceGame.java [hunk 2]
```diff
@@ -1,5 +1,4 @@
 package spacegame3;
-
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java:2]: insert back the empty line here please, it just ease the readability of the code for me :)

File: src/spacegame3/SpaceGame.java [hunk 3]
```diff
@@ -16,7 +15,8 @@
 import java.util.logging.LogManager;
 import java.util.logging.Logger;
 
-public final class SpaceGame extends Application {
+public final class SpaceGame extends Application
+{
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: please leave the opening brackets at the end of the line, it is easier for me to read and takes less lines in the file

normally when you modify other people's code, try to follow their coding style and format

File: src/spacegame3/SpaceGame.java [hunk 4]
```diff
@@ -42,22 +45,33 @@
 
     private boolean gameStarted;
 
-    public SpaceGame() {
+    /*
+     *SpaceGame constructor
+     *links startScreen and planetScreen to this context
+     */
+    public SpaceGame()
+    {
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: when commenting methods, constructions and fields, it is better to use /** */ instead of /* */ as it will add those comments/descriptions in the javadoc, also you don't have to specify that it is a constructor, the code is already telling that

File: src/spacegame3/SpaceGame.java [hunk 5]
```diff
@@ -42,22 +45,33 @@
...
         startScreen = new StartScreen(this);
         planetScreen = new PlanetScreen(this);
         gameScheme = null;
         gameStarted = false;
     }
 
+    //game scheme getter
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java:55]: use /** */ for methods (just repeating for each one), also don't need to specify that it is a getter, the method name gives this information, but it is interesting to tell about side effects, checks, etc., you could also add what this specific field is about (in a few words/lines)

File: src/spacegame3/SpaceGame.java [hunk 6]
```diff
@@ -42,22 +45,33 @@
...
     public GameScheme getGameScheme() {
         return gameScheme;
     }
 
+    //game scheme setter
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: use /** */ for methods (just repeating for each one), also don't need to specify that it is a setter, the method name gives this information, but it is interesting to tell about side effects, checks, etc.,
For example, in this case, it might be interesting to say that the ImageLibrary has its context set to the gameScheme StoryTelling path

File: src/spacegame3/SpaceGame.java [hunk 7]
```diff
@@ -42,22 +45,33 @@
...
     public void setGameScheme(GameScheme gameScheme) {
         this.gameScheme = gameScheme;
         ImageLibrary.setStoryTellingPath(gameScheme.getStoryPath());
     }
 
+    /*
+        *starts up the the game
+        * creates title and scene + shows
+    */
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: use /** */ for methods (just repeating for each one)

File: src/spacegame3/SpaceGame.java [hunk 8]
```diff
@@ -80,29 +94,45 @@ public void giveSceneTo(SizableScene scene){
         stage.setScene(scene);
     }
 
+    /*
+        *main method - runs + launches game
+    */
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: use /** */ for methods (just repeating for each one)

One interesting piece of info that can be added to the description of the main method is the list of arguments it can take (when the method takes some...)

File: src/spacegame3/SpaceGame.java [hunk 9]
```diff
@@ -80,29 +94,45 @@ public void giveSceneTo(SizableScene scene){
...
     public static void main(String[] args) {
         launch(args);
     }
 
+    //getter for game started
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: use /** */ for methods (just repeating for each one), also don't need to specify that it is a getter

File: src/spacegame3/SpaceGame.java [hunk 10]
```diff
@@ -80,29 +94,45 @@ public void giveSceneTo(SizableScene scene){
...
     public boolean gameStarted() {
         return gameStarted;
     }
 
-
+    /*
+        *sets the planet screen to have landed on the parameter (a CelestialBody object)
+        *returns planet screen
+    */
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: use /** */ for methods (just repeating for each one)
this description tells what it does that cant be determine from the method name which is good, maybe just change the end for "on the CelestialBody" or "on the given CelestialBody"

File: src/spacegame3/SpaceGame.java [hunk 11]
```diff
@@ -80,29 +94,45 @@ public void giveSceneTo(SizableScene scene){
...
     public SizableScene getPlanetScreen(CelestialBody planet) {
         planetScreen.setLandedOn(planet);
         return planetScreen;
     }
 
+    /*
+        *saves the current scene
+        *transfers to the start screen
+    */
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: use /** */ for methods (just repeating for each one)
also just change the word saves for stores
for me saves has a more long term sense than what is done here

File: src/spacegame3/SpaceGame.java [hunk 12]
```diff
@@ -80,29 +94,45 @@ public void giveSceneTo(SizableScene scene){
...
     public void showStartScreen(SizableScene scene) {
         previousScene = scene;
         giveSceneTo(startScreen);
     }
 
+    //gets previous scene
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: use /** */ for methods (just repeating for each one)

File: src/spacegame3/SpaceGame.java [hunk 13]
```diff
@@ -80,29 +94,45 @@ public void giveSceneTo(SizableScene scene){
...
     public SizableScene previousScene(){
         return previousScene;
     }
 
+    /*
+        *will send back to current screen if that is not null
+        * if it is null,
+    */
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java]: use /** */ for methods (just repeating for each one)

also this is not what this method does ;) i am giving you another try before i give the answer :)

File: src/spacegame3/SpaceGame.java [hunk 14]
```diff
@@ -31,6 +31,9 @@
     }
 
 
+    /*
+        *Instance variables
+    */
```
Comment (hellminister) [inline src/spacegame3/SpaceGame.java:32]: You can use /** */ on each field to say what they represents/does if it is important for outsiders to know about those fields (since it will go in the javadoc) which in this case i dont think is necessary or  you could use // either before the field or at the end of its line to say the same for developers so they what what use the field has

**Review by hellminister — CHANGES_REQUESTED** [review]

File: .idea/misc.xml (hunk 1 above)
Comment (Grace0Hud) [reply .idea/misc.xml]: I fixed it by copying and pasting from master, it should be the exact same as your original project.
